### ISSUE TITLE
Proposal to update the API

### ISSUE THREAD

**Issue by majek** [jmesnil/stomp-websocket#12]
Jeff, thanks for the excellent library!

Regarding [you comment](https://www.rabbitmq.com/blog/2012/05/14/introducing-rabbitmq-web-stomp/#comment-14859), it would be better IMO if the API instead of this:

```
Stomp.WebSocketClass = SockJS;
client = Stomp.client(url);
```

would look like:

```
var ws = new WebSocket(url); // or SockJS or whatever
client = Stomp.client(ws);
```

This would not only remove the need to establish websocket connection within your code, but also would allow a bit more flexibility. Take a look at [this blog post](https://www.rabbitmq.com/blog/2012/02/23/how-to-compose-apps-using-websockets/) where I try to explain how to compose higher level abstractions out of simple websocket-like objects.

In other words: it would be cool if Stomp client could be run on the "multiplexer" objects, not only real websockets.
